src/cli/Makefile.am

```diff
@@ -28,23 +28,43 @@
 
 include $(abs_top_nlbuild_autotools_dir)/automake/pre.am
 
-lib_LIBRARIES                       = libopenthread-cli.a
+lib_LIBRARIES                       = \
+    libopenthread-ftd-cli.a           \
+    libopenthread-mtd-cli.a           \
+    $(NULL) 
 
-libopenthread_cli_a_CPPFLAGS =        \
+CPPFLAGS_COMMON                     = \
     -I$(top_srcdir)/include           \
     -I$(top_srcdir)/src               \
     -I$(top_srcdir)/src/core          \
     $(OPENTHREAD_TARGET_DEFINES)      \
     $(NULL)
 
-libopenthread_cli_a_SOURCES         = \
+SOURCES_COMMON                      = \
     cli.cpp                           \
     cli_console.cpp                   \
     cli_dataset.cpp                   \
     cli_uart.cpp                      \
     cli_udp.cpp                       \
     $(NULL)
 
+libopenthread_ftd_cli_a_CPPFLAGS    = \
+    $(CPPFLAGS_COMMON)                \
+    -DOPENTHREAD_FTD_CLI              \
```

Comment:
You can define this in [libopenthread-cli.vcxproj](https://github.com/openthread/openthread/blob/master/etc/visual-studio/libopenthread-cli.vcxproj#L42) by adding `OPENTHREAD_FTD_CLI;`.